### PR TITLE
@KotlinCleanup("iteration with i") in Preferences.kt done

### PR DESCRIPTION
**Pull request by UvrajSB**
## Purpose / Description
KotlinCleanup("iteration with i") in Preferences.kt

## Fixes
Part of #10489

## Approach
Converted the map into list with elements as (key, value) and then used "items.toList()[i - 1].first" to obtain key and "items.toList()[i - 1].second" to obtain its value and then used it in for loop with "i" as iterator.

## How Has This Been Tested?
I have tried to create similar code on Kotlin Playground to test it
image

image

## Checklist
_Please, go through these checks before submitting the PR._

- [x] You have not changed whitespace unnecessarily (it makes diffs hard to read)
- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] Your code follows the style of the project (e.g. never omit braces in `if` statements) 
- [ ] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [ ] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

File: AnkiDroid/src/main/java/com/ichi2/anki/Preferences.kt [hunk 1]
```diff
@@ -733,11 +732,9 @@ class Preferences : AnkiActivity() {
                 val languageDialogValues = arrayOfNulls<CharSequence>(items.size + 1)
                 languageDialogLabels[0] = resources.getString(R.string.language_system)
                 languageDialogValues[0] = ""
-                var i = 1
-                for ((key, value) in items) {
-                    languageDialogLabels[i] = key
-                    languageDialogValues[i] = value
-                    i++
+                for (i in 1..items.size) {
+                    languageDialogLabels[i] = items.toList()[i - 1].first
```
Comment (david-allison) [inline AnkiDroid/src/main/java/com/ichi2/anki/Preferences.kt:736]: Calling toList on each iteration is inefficient. Extract it outside the loop if it's immutable

**Review by david-allison — CHANGES_REQUESTED** [review]

File: AnkiDroid/src/main/java/com/ichi2/anki/Preferences.kt (hunk 1 above)
Comment (UvrajSB) [reply AnkiDroid/src/main/java/com/ichi2/anki/Preferences.kt]: Yes, keeping it outside will make it actually much more efficient.

Comment (UvrajSB) [reply AnkiDroid/src/main/java/com/ichi2/anki/Preferences.kt]: Also, I checked this change also on Kotlin Playground, it works 
image

image

File: AnkiDroid/src/main/java/com/ichi2/anki/Preferences.kt [hunk 2]
```diff
@@ -733,11 +732,10 @@ class Preferences : AnkiActivity() {
                 val languageDialogValues = arrayOfNulls<CharSequence>(items.size + 1)
                 languageDialogLabels[0] = resources.getString(R.string.language_system)
                 languageDialogValues[0] = ""
-                var i = 1
-                for ((key, value) in items) {
-                    languageDialogLabels[i] = key
-                    languageDialogValues[i] = value
-                    i++
+                val itemsList = items.toList()
+                for (i in 1..items.size) {
```
Comment (david-allison) [inline AnkiDroid/src/main/java/com/ichi2/anki/Preferences.kt:736]: Doesn't this skip the final element?

Typically you want `itemsList.size`

**Review by david-allison — CHANGES_REQUESTED** [review]

File: AnkiDroid/src/main/java/com/ichi2/anki/Preferences.kt (hunk 2 above)
Comment (UvrajSB) [reply AnkiDroid/src/main/java/com/ichi2/anki/Preferences.kt:736]: 1) I don't think that it skips the last element 
2) The value of itemsList.size will be equal to items.size
image

Comment (david-allison) [reply AnkiDroid/src/main/java/com/ichi2/anki/Preferences.kt:736]: You're right - first one is fine.

For the second one: it's good practice to use the collection that you're iterating over in a loop

Comment (UvrajSB) [reply AnkiDroid/src/main/java/com/ichi2/anki/Preferences.kt]: Understood, change made!!

**Comment by UvrajSB** [comment]
@david-allison are any further changes required in this PR?

**Review by david-allison — APPROVED** [review]
Awesome!

**Review by ShridharGoel — APPROVED** [review]

**Comment by mikehardy** [comment]
@krmanik I love your github bot, thanks again for doing that

**Pull request merged**